satpy/readers/core/utils.py

```diff
@@ -353,7 +353,8 @@ def generic_open(filename, *args, **kwargs):
         path = filename.to_upath()
     except AttributeError:
         # str or Path-like
-        path = UPath(filename)
+        protocol = getattr(filename, "protocol", "file")
```

Comment:
The comment above this line says that this block of code is for handling `str or Path-like` but these don't have a `protocol` attribute. So what are we handling here? `UPath` objects? In that case, should we be recreating the `UPath` object?

Reply:
Previously this logic made a lot of sense as it was very duck-typing friendly (whatever UPath can handle we can handle), but now it isn't as pretty. I know you prefer the try/except pattern for stuff like this, but this may be a case for an `if isinstance` check on UPath. Or maybe a `hasattr` for something that UPath has. Or maybe a `isinstance(x, (str, os.PathLike))`...eh that would probably catch `UPath` though.